-Wduplicated-cond Warnings from Kokkos_Error.hpp

I am getting a lot of the following warnings:
~~~
kokkos/core/src/impl/Kokkos_Error.hpp: In member function ‘virtual const char* Kokkos::Experimental::RawMemoryAllocationFailure::what() const’:
kokkos/core/src/impl/Kokkos_Error.hpp:135:12: warning: duplicated ‘if’ condition [-Wduplicated-cond]
     } else if (m_failure_mode == FailureMode::OutOfMemoryError) {
            ^~
kokkos/core/src/impl/Kokkos_Error.hpp:133:5: note: previously used here
     if (m_failure_mode == FailureMode::OutOfMemoryError) {
~~~

I am compiling Kokkos as a subdirectory with OpenMP enabled, CMake 3.12.2 and gcc 7.3.0

Is there a reason for this duplicated condition or is it a bug?

--- a/core/src/impl/Kokkos_Error.hpp
+++ b/core/src/impl/Kokkos_Error.hpp
@@ -132,7 +132,7 @@ class RawMemoryAllocationFailure : public std::bad_alloc {
   const char *what() const noexcept override {
     if (m_failure_mode == FailureMode::OutOfMemoryError) {
       return "Memory allocation error: out of memory";
-    } else if (m_failure_mode == FailureMode::OutOfMemoryError) {
+    } else if (m_failure_mode == FailureMode::AllocationNotAligned) {
       return "Memory allocation error: allocation result was under-aligned";
     }
 

